text/0011-html_viz.md

```diff
@@ -0,0 +1,182 @@
+- Feature Name: html_viz
+- Start Date: 2018-08-14
+- RFC PR: [#13](https://github.com/qri-io/rfcs/pull/13)
+- Issue: <!-- (leave this empty) -->
+
+_Note: We never properly finished the HTML rendering RFC, which we started work on in August 2018.
+Instead of creating a new one we just "finished what we started" in March 2019. As such this RFC
+contains references to RFCS developed in the period between August 2018 & March 2019.
+-:heart: the qri core team_
+
+# Summary
+[summary]: #summary
+
+HTML vizualizations are instructions embedded in a dataset rendering a dataset as a standard HTML document. 
+
+# Motivation
+[motivation]: #motivation
+
+Qri has a syntax-agnostic `viz` component that encapsulates the details required to visualize a dataset. This RFC proposes the first & default visualization syntax should be rendering to a single HTML document using a template processing engine.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+### `qri render` & default template
+
+A command will be added to both Qri's HTTP API ('API' for short) & command-line interface (CLI) called `render`, which adds the capicty to execute HTML templates against datasets. When called with a specified dataset `qri render` will load the dataset, assume the viz syntax is HTML, and use a default template to write an HTML representation of a dataset to `stdout` on the CLI, or the HTTP response body on the API:
+`qri render me/example_dataset`
+
+The default template and output path can be overridden with the `--template` and `--output` flags respectively. The output is on the CLI only:
+`qri render --template template.html --output rendered.html me/example_dataset`
+
+The default template must be carefully composed to balance the size of the resulting HTML file in bytes against readability & utility of the resulting visualization. It should also include a well-constructed citation footer that details the components of a dataset in a concise, unobtrusive manner that invites users to audit the veracity of the dataset in question. These defaults should encourage easy reading and invite verification on the part of the reader.
+
+### Vizualizations in datasets
+
+Saving a dataset will by default execute the default template to a file called `index.html` & place it in the IPFS merkle-DAG. When an IPFS HTTP gateway receives a request for a DAG that is a directory containing `index.html`, it returns the HTML file by default. This means when a user visits a dataset on the d.web–completely outside the Qri system of dataset production–they are greeted with a well-formatted dataset document by default.
```

Comment:
This is super clever. Do we know for sure that IPFS gateways serve index.html instead of showing a directory listing? I thought that might be an Apache specific thing. Is there a way to force the IPFS gateway to show the datasets contents?

Reply:
It's the default gateway behaviour for unixfs v1 dags. Looking at the code, I don't think there's a way to force listing directory contents: https://github.com/ipfs/go-ipfs/blob/74d07eff35965a3f635d03aedaa43561c73679e2/core/corehttp/gateway_handler.go#L249-L274

That seems like it would be useful. Might make for a nice PR into go-ipfs to add a query param, I have no idea if they'd accept. @lgierth might be a good person to ask 😄.

In the meantime tho, our `citation` partial should list a relative link to the `/dataset.json` definition, which should help a bit.

Reply:
Yeah, linking to the dataset.json file should work fine, since it functions as a self-describing doc of the dataset.